### PR TITLE
Add support for a configuration file

### LINKED ISSUE #13: Support configuration from file
only CLI arguments are supported today.
It would improve UX if we add support for configuration file, which can be passed as CLI argument with default value.

### PR DESCRIPTION
**Pull request by pscott**
- Adds support for a configuration file (named `config.toml`).
- Introduces dependencies to [confy](https://docs.rs/confy/latest/confy/) and [serde_derive](https://crates.io/crates/serde_derive) to parse the configuration file.

The flow is:
1. Try to parse the arguments provided via the CLI.
2. If arguments were correct, continue. If arguments were provided but they're wrong, display the error and exit. If no arguments were provided, try to parse the arguments in the configuration file. 
3. If arguments were correct, continue. If arguments were provided but they're wrong, display "ERROR in your configuration file". Then display the error, regardless of whether some arguments were provided or not.

Ref: https://app.onlydust.xyz/contributions/0x005d
Fixes #13 

Time dedicated to this issue: 4h (took me some time to find a clean way of doing this... I had to dabble around a little bit with clap and  other libs for a bit).

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -4,16 +4,76 @@ use cairo_foundry::cli::{
...
+const CONFIGURATION_FILENAME: &str = "config.toml";
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+struct Config {
+	args: String,
+}
+
+/// Tries to retrieve the Args from the CLI. If the user left the CLI empty, tries to
+/// retrieve the args from the configuration file.
+fn get_args() -> cli::Args {
```
Comment (tdelabro) [inline src/main.rs:20]: Can you move it in a different file, in order not to cluster main.rs

File: src/main.rs [hunk 2]
```diff
@@ -4,16 +4,76 @@ use cairo_foundry::cli::{
...
+	// Try to parse arguments from stdin
+	let args_from_stdin: Result<cli::Args, clap::Error> = cli::Args::try_parse();
+
+	match args_from_stdin {
+		Err(stdin_err) => {
+			// Rust trick to help with type system
+			let stdin_err: clap::Error = stdin_err;
```
Comment (tdelabro) [inline src/main.rs:27]: Why is this necessary?
Can you explain in more details? Here or in the comment itself

File: src/main.rs [hunk 3]
```diff
@@ -4,16 +4,76 @@ use cairo_foundry::cli::{
...
+
+			match stdin_err.kind() {
+				// This means the user left the CLI empty. We can try parsing the configuration file
+				clap::ErrorKind::DisplayHelpOnMissingArgumentOrSubcommand => {
```
Comment (tdelabro) [inline src/main.rs:31]: You are only handling one case, so maybe an `if let / else` would be more expressive there

**Review by tdelabro — CHANGES_REQUESTED** [review]
In your implementation, the arguments executed are either all from the CLI or all from the config file.

What we had in mind was more: for each argument, use the CLI if provided, otherwise, go look for it in the config file.
Do you think something like this could be done ?

File: src/main.rs (hunk 1 above)
Comment (pscott) [reply src/main.rs:20]: Yeah didn't know what the guideline were for `cairo-foundry`: will do just that :)

File: src/main.rs (hunk 2 above)
Comment (pscott) [reply src/main.rs:27]: Not necessary but helps with `rust-analyzer` and other tools help understand what's going on :D

File: src/main.rs (hunk 1 above)
Comment (pscott) [reply src/main.rs]: ae3a980947f4ee64a932610637d6c54aadd24afa

File: src/main.rs (hunk 2 above)
Comment (pscott) [reply src/main.rs]: 6f1f899fa942ce8a8fbcd0d23466f33d89675d3f

File: src/main.rs (hunk 3 above)
Comment (pscott) [reply src/main.rs]: https://github.com/onlydustxyz/cairo-foundry/commit/ae3a980947f4ee64a932610637d6c54aadd24afa
